.github/workflows/python-tests.yml

```diff
@@ -26,6 +26,8 @@ jobs:
             python-version: "pypy-3.8"
           - os: macos-latest
             python-version: "3.10"
+          - os: ubuntu-latest
+            python-version: "3.12.0-beta.1"
```

Comment:
I learned [from here](https://dev.to/hugovk/help-test-python-312-beta-1508#github-actions-setuppython) that you can use `python-version: "3.12"` if you add `allow-prereleases: true` to setup-python. Then we don't have to change this field as 3.12 move towards release.